### ISSUE TITLE
CallablePublicOnlyProxy creates trouble with numpy

### ISSUE THREAD

**Issue by gatoniel** [napari/napari#3774]
## 🐛 Bug

Hi,
I don't know when and how a `layer.data` gets the type `napari.utils._proxies.CallablePublicOnlyProxy` instead of `numpy.ndarray`. But when it has this type it cannot directly be passed to skimage as the data type cannot be interpreted by numpy.

## To Reproduce

I am sorry, but I did not find any way to reproduce the error without using the Widget Plugins.

1. Run the following code in an interactive ipython console with the napari development version 0.4.13.dev112+ge6ce01d9 (current main branch)
```
from napari import Viewer
from tifffile import imread

from skimage.measure import label
from skimage.data import cells3d
from napari_3d_ortho_viewer._dock_widget import OrthoViewerWidget


img = cells3d()[30:52, 1]
lbl = label(img > img.mean())

viewer = Viewer()
img_layer = viewer.add_image(img)
lbl_layer = viewer.add_labels(lbl)

print(lbl_layer.data.dtype)
print(type(lbl_layer.data))
lbl_layer.as_layer_data_tuple()
print(lbl_layer.data.dtype)
print(type(lbl_layer.data))

widget = OrthoViewerWidget(viewer)

viewer.window.add_dock_widget(widget)
```
2. Click `Start/Stop Ortho viewer` to start ortho viewer.
3. Click `Start/Stop Ortho viewer` to stop ortho viewer again.
4. In the ipython console inspect the type of the `lbl_layer.data` with `type(lbl_layer.data)`. It is `numpy.ndarray`.
5. Plugins -> napari-3d-ortho-viewer -> Crop Labels Widget
6. Observe that within the plugin the type of `layer.data` is `napari.utils._proxies.CallablePublicOnlyProxy` this creates the following error when calling `regionprops` on it:
```
TypeError                                 Traceback (most recent call last)
d:\napari-3d-ortho-viewer\.win-napari-dev-3d-ortho-viewer\lib\site-packages\napari\_qt\qt_main_window.py in add_plugin_dock_widget(self, plugin_name, widget_name)
    673             return dock_widget, wdg
    674
--> 675         wdg = _instantiate_dock_widget(Widget, self._qt_viewer.viewer)
    676
    677         # Add dock widget

d:\napari-3d-ortho-viewer\.win-napari-dev-3d-ortho-viewer\lib\site-packages\napari\_qt\qt_main_window.py in _instantiate_dock_widget(wdg_cls, viewer)
   1214
   1215     # instantiate the widget
-> 1216     return wdg_cls(**kwargs)

d:\napari-3d-ortho-viewer\src\napari_3d_ortho_viewer\_crop_dock_widget.py in __init__(self, viewer)
     46
     47         self.id_selection = None
---> 48         self.create_id_selection()
     49
     50     def create_id_selection(self) -> None:

d:\napari-3d-ortho-viewer\src\napari_3d_ortho_viewer\_crop_dock_widget.py in create_id_selection(self)
     54
     55         self.id_selection = Select(
---> 56             name="labels_ids_dropdown", label="Label IDs", choices=self.labels_list()
     57         )
     58         self.layout().addWidget(self.id_selection.native)

d:\napari-3d-ortho-viewer\src\napari_3d_ortho_viewer\_crop_dock_widget.py in labels_list(self)
     75         print(selected_layer.data.dtype)
     76         print(type(selected_layer.data))
---> 77         self.regions = regionprops(selected_layer.data)
     78         area = np.array([reg.area for reg in self.regions])
     79

d:\napari-3d-ortho-viewer\.win-napari-dev-3d-ortho-viewer\lib\site-packages\skimage\measure\_regionprops.py in regionprops(label_image, intensity_image, cache, coordinates, extra_properties)
   1251         raise TypeError('Only 2-D and 3-D images supported.')
   1252
-> 1253     if not np.issubdtype(label_image.dtype, np.integer):
   1254         if np.issubdtype(label_image.dtype, bool):
   1255             raise TypeError(

d:\napari-3d-ortho-viewer\.win-napari-dev-3d-ortho-viewer\lib\site-packages\numpy\core\numerictypes.py in issubdtype(arg1, arg2)
    416     """
    417     if not issubclass_(arg1, generic):
--> 418         arg1 = dtype(arg1).type
    419     if not issubclass_(arg2, generic):
    420         arg2 = dtype(arg2).type

TypeError: Cannot interpret 'dtype('int32')' as a data type
```

If you do the same with the current stable release no error is produced. Furthermore, if you leave out steps 3) and 4) no error occurs and the type of `layer.data` within the widget stays `numpy.ndarray`.

## Expected behavior

I would expect that running above steps in the current main branch does not create an error.

### stable napari:
napari: 0.4.12
Platform: Windows-10-10.0.22000-SP0
Python: 3.9.5 (tags/v3.9.5:0a7dcbd, May 3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.6
NumPy: 1.21.4
SciPy: 1.7.3
Dask: 2021.11.2
VisPy: 0.9.4

OpenGL:
- GL version: 4.6.0 - Build 27.20.100.9664
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 2560x1440, scale 1.0
- screen 2: resolution 2560x1440, scale 1.0

Plugins:
- console: 0.0.4
- napari-3d-ortho-viewer: 0.0.1
- scikit-image: 0.4.12
- svg: 0.1.5

### development napari:
napari: 0.4.13.dev112+ge6ce01d9
Platform: Windows-10-10.0.22000-SP0
Python: 3.9.5 (tags/v3.9.5:0a7dcbd, May 3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.6
NumPy: 1.21.4
SciPy: 1.7.3
Dask: 2021.11.2
VisPy: 0.9.4

OpenGL:
- GL version: 4.6.0 - Build 27.20.100.9664
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 2560x1440, scale 1.0
- screen 2: resolution 2560x1440, scale 1.0

Plugins:
- console: 0.0.4
- napari-3d-ortho-viewer: 0.0.1
- scikit-image: 0.4.13.dev112+ge6ce01d9
- svg: 0.1.5

 - Any other relevant information:

## Additional context

I used the current main branch of https://github.com/gatoniel/napari-3d-ortho-viewer/commit/173cb09e502bfeef6a13dae780782765cb123b7a

**Comment by tlambert03** [comment]
thanks for reporting!  will get this fixed

**Comment by tlambert03** [comment]
fix in #3777

**Comment by gatoniel** [comment]
I tested it on #3777 and the error is gone. Thanks!
